src/main/scala/uk/gov/nationalarchives/tdr/api/http/Routes.scala

```diff
@@ -30,6 +34,33 @@ class Routes(val config: Config) extends Cors {
   val transportSecurityMaxAge = 31536000
   implicit val keycloakDeployment: TdrKeycloakDeployment = TdrKeycloakDeployment(url, "tdr", ttlSeconds)
 
+  val exceptionHandler: ExceptionHandler =
+    ExceptionHandler {
+      case ex: Throwable =>
+        entity(as[JsValue]) { requestJson =>
+          val JsObject(fields) = requestJson
+          val JsString(query) = fields("query")
+
+          QueryParser.parse(query) match {
+            case Failure(parseException) =>
+              logger.error(s"Error parsing entity $requestJson", parseException)
+              complete(HttpResponse(InternalServerError, entity = parseException.getMessage))
```

Comment:
It looks like we're returning the error about the parse exception, but we're not including any information about the original exception. Am I following that right? Would it be better to return an internal server error based on the original error, since the parsing was just done to get some more context?

Reply:
Yeah we're returning the parse exception but it would make more sense to have the original exception. I've left the log line for the parse exception because it might be useful. I've added in a log line for the original exception which just includes the message and returned the exception message in the response.